scripts/o3de/o3de/repo_properties.py

```diff
@@ -0,0 +1,377 @@
+                    elif repo_objects_data[index].get('version','0.0.0') == version_data['version']:
+                        repo_objects_data[index].update(version_data)
+                    else:    
+                        # add the new version
+                        if 'versions_data' not in repo_objects_data[index]:
+                            repo_objects_data[index]['versions_data'] = [version_data] 
+                        else:
+                            repo_objects_data[index]['versions_data'].append(version_data) 
+                else:
+                    repo_objects_data.append(json_data)
+
+    if delete_objects:      
+        removal_list = delete_objects.split() if isinstance(delete_objects, str) else delete_objects
+        # find name field of object you want to delete
+        object_key = str(f'{object_typename}_name')
+        
+        for removal_object in removal_list:
+            # Remove the JSON object(s) with the specified object_name value
+            object_name, version_specifier = utils.get_object_name_and_optional_version_specifier(removal_object)
+            if not version_specifier:
+                # Remove the JSON object(s) with the specified object_name value
+                repo_objects_data = [object_typename for object_typename in repo_objects_data if object_typename.get(object_key) != object_name]
+            else:
+               index = _find_index(repo_objects_data, f'{object_typename}_name', object_name)
+               #if version exists 
+               if index > -1:
+                   # returns the version data field associated object version data with input, if not return version data return empty field
+                   versions_data = repo_objects_data[index].get('versions_data',[])
+                   if versions_data:
+                       # remove matching versions
+                       versions_data = [data for data in versions_data if Version(data.get('version', '0.0.0')) not in SpecifierSet(version_specifier)]
+                       repo_objects_data[index]['versions_data'] = versions_data
+                              
+                   # remove/replace base version if it matches
+                   if Version(repo_objects_data[index].get('version', '0.0.0')) in SpecifierSet(version_specifier):
+                       if versions_data:
+                            # updates the base version with newer version
+                            repo_objects_data[index].update(versions_data.pop(0))
+                            repo_objects_data[index]['versions_data'] = versions_data
+                       else: 
+                            # versions data is empty so remove the whole object entry
+                            del repo_objects_data[index]              
+
+    repo_json[f'{object_typename}s_data'] = repo_objects_data
+     #if older version is removed, merged the items thats changed in the newer version if newer version exist and update version
```

Comment:
I don't think this comment is in the correct place any more?
```suggestion
```

Reply:
yes, this should just be removed, sorry about that